sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveUtilsSuite.scala

```diff
@@ -42,4 +47,29 @@ class HiveUtilsSuite extends QueryTest with SQLTestUtils with TestHiveSingleton
       assert(hiveConf("foo") === "bar")
     }
   }
+
+  test("ChildFirstURLClassLoader's parent is null") {
+    val conf = new SparkConf
+    val contextClassLoader = Thread.currentThread().getContextClassLoader
+    val loader = new FakeChildFirstURLClassLoader(Array(), contextClassLoader)
+    Thread.currentThread().setContextClassLoader(loader)
+    intercept[IllegalArgumentException](
+      HiveUtils.newClientForMetadata(conf, SparkHadoopUtil.newConfiguration(conf)))
+    Thread.currentThread().setContextClassLoader(contextClassLoader)
+  }
+
+  test("ChildFirstURLClassLoader's parent is null, get spark classloader instead") {
+    val conf = new SparkConf
+    val contextClassLoader = Thread.currentThread().getContextClassLoader
+    val loader = new ChildFirstURLClassLoader(Array(), contextClassLoader)
+    Thread.currentThread().setContextClassLoader(loader)
+    HiveUtils.newClientForMetadata(conf, SparkHadoopUtil.newConfiguration(conf))
+    Thread.currentThread().setContextClassLoader(contextClassLoader)
+  }
 }
+
+/**
+ * A Fake [[ChildFirstURLClassLoader]] used for test
+ */
+private[spark] class FakeChildFirstURLClassLoader(urls: Array[URL], parent: ClassLoader)
```

Comment:
This is not a `ChildFirstURLClassLoader`, shall we make it extend `ChildFirstURLClassLoader`?

Reply:
inheritation maketh case match